core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
Have a look at my last commit f3847c6, I changed the builder so it renders out a query even if one of the clauses was not set. This will produce queries like: `{
  "boosting" : {
    "positive" : {
      "match_all" : { }
    },
    "negative_boost" : 0.1,
    "boost" : 1.0
  }
}`

which will be rejected by the parser because there we require the user to state both clauses. It's hard to render a query that will parse but will parse to `null`, I think `{}` doesn't work here and just trying to render the fieldname results in XContentBuilder errors for me. 
The core problem is that we need the builders to allow something when they serve as the "intermediate query representation" (namely null clauses) that they shouldn't support in their original use as json-query builders. Given that we eventually get rid of doXContent here I'd be happy to go this way.